open-sauced-admin/action.js

```diff
@@ -17,46 +17,47 @@ async function run() {
       return
     }
     
-    // for debugging 
-    // if (repository.full_name !== "bdougie/open-sauced-goals") {
-    //   return
-    // }
-
     const {data} = await octokit.rest.repos.getContent({
       owner: repository.owner.login,
       repo: repository.name,
       path: "data.json"
-    })
+    }).catch((err) => {
+      console.log(`data.json: ${err}`)
+      return {data: {content: []}}
+    });
 
-    // convert from base64 to parseable JSOON 
-    // (replace with octokit/plugin-create-or-update-text-file.js)
-    const content = Buffer.from(data.content, "base64").toString()
-    const parsedData = JSON.parse(content)
+    try {
+      // convert from base64 to parseable JSOON 
+      const content = Buffer.from(data.content, "base64").toString()
+      const parsedData = JSON.parse(content)
 
-    // update repo stats
-    for (const item of parsedData) {
-      const [owner, repo] = item.full_name.split("/")
-      const currentRepoResponse = await octokit.rest.repos.get({owner, repo})
-      item.stargazers_count = currentRepoResponse.data.stargazers_count
-      item.forks_count = currentRepoResponse.data.forks_count
-      item.open_issues_count = currentRepoResponse.data.open_issues_count
+      // update repo stats
+      for (const item of parsedData) {
+        const [owner, repo] = item.full_name.split("/")
+        const currentRepoResponse = await octokit.rest.repos.get({owner, repo})
+        item.stargazers_count = currentRepoResponse.data.stargazers_count
+        item.forks_count = currentRepoResponse.data.forks_count
+        item.open_issues_count = currentRepoResponse.data.open_issues_count
+      }
+      
+      const dataString = JSON.stringify(parsedData, null, 2)
+      const base64String = Buffer.from(dataString).toString("base64")
+      
+      // only make commit if there are changes
+      await octokit.rest.repos.createOrUpdateFileContents({
+        owner: repository.owner.login,
+        repo: repository.name, 
+        path: "data.json", 
+        content: base64String, 
+        message: "updated goal data", 
+        sha: data.sha 
+      })
+      
+      console.log(`UPDATED: ${repository.html_url}`)
+    } catch {
+      console.log(`SKIPPED: ${repository.html_url}`)
```

Comment:
Adding this note to easily folks who have not added any data to their project. Might be worth adding this to the admin dashboard (how many users don't have data).